onnx/optimizer/passes/fuse_bn_into_conv.h

```diff
@@ -0,0 +1,179 @@
+    auto W_iter = graph.getInitializer(conv_inputs[1]->uniqueName());
+    if (s_iter == end_iter || bbn_iter == end_iter || m_iter == end_iter ||
+        var_iter == end_iter || W_iter == end_iter) {
+      return false;
+    }
+
+    ONNX_ASSERT(s_iter->sizes().size() == 1);
+    ONNX_ASSERT(bbn_iter->sizes().size() == 1 && bbn_iter->sizes()[0] == s_iter->sizes()[0]);
+    ONNX_ASSERT(m_iter->sizes().size() == 1 && m_iter->sizes()[0] == s_iter->sizes()[0]);
+    ONNX_ASSERT(var_iter->sizes().size() == 1 && var_iter->sizes()[0] == s_iter->sizes()[0]);
+    ONNX_ASSERT(W_iter->sizes().size() > 2 && W_iter->sizes()[0] == s_iter->sizes()[0]);
+    ONNX_ASSERT(s_iter->elem_type() == bbn_iter->elem_type() &&
+                s_iter->elem_type() == m_iter->elem_type() &&
+                s_iter->elem_type() == var_iter->elem_type() &&
+                s_iter->elem_type() == W_iter->elem_type());
+    if (s_iter->elem_type() != ONNX_NAMESPACE::TensorProto_DataType_FLOAT &&
+        s_iter->elem_type() != ONNX_NAMESPACE::TensorProto_DataType_DOUBLE)  {
+      return false;
+    }
+
+    Tensor bc;
+    if (conv_inputs.size() == 3) {
+      auto bc_iter = graph.getInitializer(conv_inputs[2]->uniqueName());
+      if (bc_iter == end_iter) {
+        return false;
+      }
+      bc = *bc_iter;
+      ONNX_ASSERT(bc.sizes().size() == 1 && bc.sizes()[0] == s_iter->sizes()[0]);
+    }
+
+    auto s = *s_iter;
+    auto bbn = *bbn_iter;
+    auto m = *m_iter;
+    auto var = *var_iter;
+    auto W = *W_iter;
+    float epsilon = bn->hasAttribute(kepsilon) ? (float) bn->f(kepsilon) : 1e-5f;
+    Tensor eps;
+
+    #define DO_COMPUTATION(TENSOR_TYPE, vec)                                   \
+      eps.sizes().push_back(s.sizes()[0]);                                     \
+      eps.elem_type() = ONNX_NAMESPACE::TensorProto_DataType_##TENSOR_TYPE;    \
+      for (int64_t i = 0; i < eps.sizes()[0]; ++i)  {                          \
+        eps.vec().push_back(epsilon);                                          \
+      }                                                                        \
+      if (conv_inputs.size() != 3) {                                           \
+        bc.sizes().push_back(s.sizes()[0]);                                    \
+        bc.elem_type() = ONNX_NAMESPACE::TensorProto_DataType_##TENSOR_TYPE;   \
+        for (int64_t i = 0; i < eps.sizes()[0]; ++i)  {                        \
+          bc.vec().push_back(0.f);                                             \
+        }                                                                      \
+      }                                                                        \
+      var.add(eps);                                                            \
+      var.sqrt();                                                              \
+      s.divide(var);                                                           \
```

Comment:
You need to check `s` and `var` don't have other uses in rest of the graph before modifying them inplace

Reply:
they don't because I'm copying the tensors into s and var (L99, L102) 
auto s = *s_iter;
auto var = *var_iter;